kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -542,6 +659,216 @@ namespace Kratos
+			std::vector<array_1d <double,3> > intsect_pts_vector;
+			ConvertRatiosToIntersectionPoints(r_geometry, combined_edge_ratios, intsect_pts_vector);
+
+			// If there are more than 3 intersected edges (3D), compute the least squares plane approximation
+			// by using the ComputePlaneApproximation utility. Otherwise, the distance is computed using
+			// the plane defined by the 3 intersection points (3D).
+			const bool do_plane_approx = (n_cut_edges == TDim) ? false : true;
+
+			if (do_plane_approx){
+				// Call the plane optimization utility
+				array_1d<double,3> base_pt, normal;
+				ComputePlaneApproximation(rElement, intsect_pts_vector, base_pt, normal);
+
+				// Compute the distance to the approximation plane
+				Plane3D approximation_plane(normal, Point{base_pt});
+				for (uint8_t i = 0; i < n_nodes; i++) {
+					rElementalDistancesExtraVector[i] = approximation_plane.CalculateSignedDistance(r_geometry[i]);
+				}
+			} else {
+				// Create a plane with the 3 intersection points (or 2 in 2D)
+				Plane3D plane = SetIntersectionPlane(intsect_pts_vector);
+
+				// Compute the distance to the intersection plane
+				for (uint8_t i = 0; i < n_nodes; i++) {
+					rElementalDistancesExtraVector[i] = plane.CalculateSignedDistance(r_geometry[i]);
+				}
+			}
+
+			// Correct the distance values orientation
+			CorrectDistanceOrientation(r_geometry, rIntersectedObjects, rElementalDistancesExtraVector);
+		}
+
+	}
+
+	template<std::size_t TDim>
+	void CalculateDiscontinuousDistanceToSkinProcess<TDim>::ConvertRatiosToIntersectionPoints(
+        const Element::GeometryType& rGeometry,
+        const array_1d<double, (TDim == 2) ? 3 : 6> &rEdgeRatiosVector,
+        std::vector<array_1d <double,3> > &rIntersectionPointsVector)
+	{
+		constexpr std::size_t n_edges = (TDim == 2) ? 3 : 6;
+		const auto r_edges_container = rGeometry.GenerateEdges();
+
+		// Clear intersection points vector
+		rIntersectionPointsVector.clear();
+
+		// Calculate intersection point of each edge that is intersected
+		for (std::size_t i_edge = 0; i_edge < n_edges; ++i_edge){
+			if (rEdgeRatiosVector[i_edge] >= 0.0){
+				array_1d<double,3> avg_pt = ConvertEdgeRatioToIntersectionPoint(r_edges_container[i_edge], rEdgeRatiosVector[i_edge]);
+				rIntersectionPointsVector.push_back(avg_pt);
+			}
+		}
+	}
+
+	template<std::size_t TDim>
+	double CalculateDiscontinuousDistanceToSkinProcess<TDim>::ConvertIntersectionPointToEdgeRatio(
+        const Kratos::Geometry<Kratos::Node<3> >& rEdge,
+        const array_1d<double,3>& rIntersectionPoint)
+	{
+		const double edge_length = rEdge.Length();
+		const double dist_avg_pt = norm_2(rEdge[0] - rIntersectionPoint);
+		return dist_avg_pt / edge_length;
+	}
+
+	template<std::size_t TDim>
+	array_1d<double,3> CalculateDiscontinuousDistanceToSkinProcess<TDim>::ConvertEdgeRatioToIntersectionPoint(
+        const Kratos::Geometry<Kratos::Node<3> >& rEdge,
```

Comment:
```suggestion
        const Geometry<Node<3>>& rEdge,
```
You are already in the Kratos namespace so there is no need to specify it.